src/main/kotlin/app/extractors/ExtractorInterface.kt

```diff
@@ -61,24 +66,95 @@ interface ExtractorInterface {
                             outstream.close()
                         }
                     }
-
                 }
             }
             catch (e: Exception) {
                 Logger.error(e, "Failed to download $name model")
             }
         }
 
+        private fun downloadModelVersions(outputFileName: String) {
+            val url = BuildConfig.LIBRARY_MODELS_URL + outputFileName
+            val file = FileHelper.getFile(outputFileName, modelsDir)
+            val builder = HttpClientBuilder.create()
+            val client = builder.build()
+            try {
+                client.execute(HttpGet(url)).use { response ->
+                    val entity = response.entity
+                    if (entity != null) {
+                        FileOutputStream(file).use { outstream ->
+                            entity.writeTo(outstream)
+                            outstream.flush()
+                            outstream.close()
+                        }
+                    }
+                }
+            }
+            catch (e: Exception) {
+                Logger.error(e, "Failed to download versions file")
+            }
+        }
+
+        @Throws(InvalidProtocolBufferException::class)
+        fun getModelsVersions(proto: ModelsVersions):
+            Map<String, Int> {
+            return proto.languagesList.zip(proto.versionList).toMap()
+        }
+
         fun getLibraryClassifier(name: String): Classifier {
+            val fileName = name + pbExt
+
             if (classifiersCache.containsKey(name)) {
                 return classifiersCache[name]!!
             }
 
-            val fileName = name + pbExt
-            if (FileHelper.notExists(fileName, modelsDir)) {
+            val appVersionFileName = "app_versions.pb"
+            val storageVersionFileName = "versions.pb"
+            if (FileHelper.notExists(appVersionFileName, modelsDir)) {
```

Comment:
Code for downloading and checking model version is better to move to different functions, e.g.:
```
fun getLibraryClassifier(name: String): Classifier {
  if (cache.containsKey(name)) { return cache[name] }
  if (FileHelper.notExists(name, modelsDir) || !isLastestModelVersion(name)) {
    downloadModel(name)  // This function should also update app model version file.
  }
  val model = readModel(name)
  cache[name] = model
  return model
}
```